src/cascadia/TerminalApp/AzureCloudShellGenerator.h

```diff
@@ -0,0 +1,36 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- AzureCloudShellGenerator
+
+Abstract:
+- This is the dynamic profile generator for the azure cloud shell connector.
+  Checks if the Azure Cloud shell is available on this playform, and if it is,
+  creates a profile to be able to launch it.
+
+Author(s):
+- Mike Griese - August 2019
+
+--*/
+
+#pragma once
+#include "IDynamicProfileGenerator.h"
+
+static constexpr GUID AzureConnectionType = { 0xd9fcfdfa, 0xa479, 0x412c, { 0x83, 0xb7, 0xc5, 0x64, 0xe, 0x61, 0xcd, 0x62 } };
+
+namespace TerminalApp
+{
+    class AzureCloudShellGenerator;
+};
+
+class TerminalApp::AzureCloudShellGenerator : public TerminalApp::IDynamicProfileGenerator
+{
+public:
+    AzureCloudShellGenerator() = default;
+    ~AzureCloudShellGenerator() = default;

```

Comment:
Since we are defaulting these we should default / delete the rest of the defaulted methods.  CppCoreGuide line [C21](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-default-operation-define-or-delete-them-all)